### PR TITLE
Fix the dns record creation/deletion if GKE plans

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Fix the DNS record creation/deletion if GKE plans

**Related issue(s)**

- https://github.com/kyma-project/test-infra/issues/525

### CODE REVIEW

File: prow/scripts/cluster-integration/helpers/create-dns-record.sh [hunk 1]
```diff
@@ -25,23 +25,33 @@ if [ "${discoverUnsetVar}" = true ] ; then
...
+        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
+
+        if [[ "${i}" -lt "${retries}" ]]; then
+            echo "Unable to create DNS record, let's wait ${retryTimeInSec} seconds and retry [${i}/${retries}]."
+        else
+            echo "Unable to create DNS record after ${retries} retries, giving up."
+        fi
 
-gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction add "${IP_ADDRESS}" --name="${DNS_FULL_NAME}" --ttl=300 --type=A --zone="${CLOUDSDK_DNS_ZONE_NAME}"
+        sleep retryTimeInSec
```
Comment (Demonsthere) [inline prow/scripts/cluster-integration/helpers/create-dns-record.sh]: Missing `$` here

File: prow/scripts/cluster-integration/helpers/delete-dns-record.sh [hunk 2]
```diff
@@ -24,10 +24,31 @@ if [ "${discoverUnsetVar}" = true ] ; then
...
-echo "DNS Record deleted, but it can be visible for some time due to DNS caches"
+        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
+
+        if [[ "${i}" -lt "${retries}" ]]; then
+            echo "Unable to delete DNS record, let's wait ${retryTimeInSec} seconds and retry [${i}/${retries}]."
+        else
+            echo "Unable to delete DNS record after ${retries} retries, giving up."
+        fi
+
+        sleep retryTimeInSec
```
Comment (Demonsthere) [inline prow/scripts/cluster-integration/helpers/delete-dns-record.sh]: Also missing `$`

File: prow/scripts/cluster-integration/helpers/create-dns-record.sh [hunk 3]
```diff
@@ -25,23 +25,33 @@ if [ "${discoverUnsetVar}" = true ] ; then
...
-CLEANUP_DNS_TRANSACTION=true
+        [[ $? -eq 0 ]] && break
 
-gcloud dns --project="${CLOUDSDK_CORE_PROJECT}" record-sets transaction start --zone="${CLOUDSDK_DNS_ZONE_NAME}"
+        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
+
+        if [[ "${i}" -lt "${retries}" ]]; then
+            echo "Unable to create DNS record, let's wait ${retryTimeInSec} seconds and retry [${i}/${retries}]."
+        else
+            echo "Unable to create DNS record after ${retries} retries, giving up."
```
Comment (jakkab) [inline prow/scripts/cluster-integration/helpers/create-dns-record.sh]: The script should exit with code 1 if it gives up. Waiting 10 minutes for something that is not going to happen seems pointless.

File: prow/scripts/cluster-integration/helpers/create-dns-record.sh [hunk 4]
```diff
@@ -25,23 +25,33 @@ if [ "${discoverUnsetVar}" = true ] ; then
...
+retryTimeInSec="5"
+function createDNSWithRetries() {
+    set +e
 
-cleanup() {
-    if [ "${CLEANUP_DNS_TRANSACTION}" == true ]; then
-        gcloud dns record-sets transaction abort --zone="${CLOUDSDK_DNS_ZONE_NAME}" --verbosity none
-    fi
-}
+    for ((i=1; i<=retries; i++)); do
```
[the remark this answers is not in the document]
Comment (mszostok) [reply prow/scripts/cluster-integration/helpers/create-dns-record.sh]: just changed the name from `retries` to `attempts`

File: prow/scripts/cluster-integration/helpers/create-dns-record.sh (hunk 3 above)
Comment (mszostok) [reply prow/scripts/cluster-integration/helpers/create-dns-record.sh]: good catch

**Pull request merged**